src/overlays/actors/ovl_Bg_Kin2_Bombwall/z_bg_kin2_bombwall.c

```diff
@@ -28,50 +40,194 @@ const ActorInit Bg_Kin2_Bombwall_InitVars = {
+
+    Actor_ProcessInitChain(&this->dyna.actor, D_80B6E748);
+    DynaPolyActor_Init((DynaPolyActor*)this, 0);
+    bombwallCollider = &this->collider;
+    Collider_InitCylinder(globalCtx, bombwallCollider);
+    if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x7F) != 0) {
+        Actor_MarkForDeath(&this->dyna.actor);
+        return;
+    }
+    DynaPolyActor_LoadMesh(globalCtx, (DynaPolyActor*)this, (CollisionHeader*)&D_06000490);
+    Collider_SetCylinder(globalCtx, bombwallCollider, &this->dyna.actor, &D_80B6E6F0);
+    Collider_UpdateCylinder(&this->dyna.actor, bombwallCollider);
+    Actor_SetFocus(&this->dyna.actor, 60.0f);
+    BgKin2Bombwall_SetupWait(this);
+}
+
+void BgKin2Bombwall_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgKin2Bombwall* this = (BgKin2Bombwall*)thisx;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void BgKin2Bombwall_SetupWait(BgKin2Bombwall* arg0) {
+    arg0->actionFunc = BgKin2Bombwall_Wait;
+}
+
+void BgKin2Bombwall_Wait(BgKin2Bombwall* arg0, GlobalContext* arg1) {
+    if (func_80B6E020(arg0, arg1) != 0) { // checks if AC collision happened
+        arg0->collider.base.acFlags &= 0xFFFD;
+        ActorCutscene_SetIntentToPlay((s16)arg0->dyna.actor.cutscene);
+        BgKin2Bombwall_SetupPlayCutscene(arg0);
+        return;
+    }
+    CollisionCheck_SetAC(arg1, &arg1->colChkCtx, &arg0->collider.base);
+}
+
+void BgKin2Bombwall_SetupPlayCutscene(BgKin2Bombwall* arg0) {
+    arg0->actionFunc = BgKin2Bombwall_PlayCutscene;
+}
+
+void BgKin2Bombwall_PlayCutscene(BgKin2Bombwall* arg0, GlobalContext* arg1) {
+    if (ActorCutscene_GetCanPlayNext((s16)arg0->dyna.actor.cutscene) != 0) {
+        ActorCutscene_StartAndSetUnkLinkFields((s16)arg0->dyna.actor.cutscene, &arg0->dyna.actor);
+        Flags_SetSwitch(arg1, arg0->dyna.actor.params & 0x7F);
+        SoundSource_PlaySfxAtFixedWorldPos(arg1, &arg0->dyna.actor.world.pos, 0x3C, 0x2810U);
+        func_800C62BC(arg1, &arg1->colCtx.dyna, arg0->dyna.bgId);
+        arg0->dyna.actor.draw = NULL;
+        func_80B6E090(arg0, arg1); // not sure yet, but takes care of the explosion effects and particles like dust.
+        BgKin2Bombwall_SetupEndCutscene(arg0);
+        return;
+    }
+    ActorCutscene_SetIntentToPlay((s16)arg0->dyna.actor.cutscene);
+}
+
+void BgKin2Bombwall_SetupEndCutscene(BgKin2Bombwall* arg0) {
+    arg0->unk_1AC[0] = 0x28;
+    arg0->actionFunc = BgKin2Bombwall_EndCutscene;
+}
+
+void BgKin2Bombwall_EndCutscene(BgKin2Bombwall* arg0, GlobalContext* arg1) {
+    arg0->unk_1AC[0]--;
+    if ((s32)arg0->unk_1AC[0] <= 0) {
+        ActorCutscene_Stop((s16)arg0->dyna.actor.cutscene);
+        Actor_MarkForDeath(&arg0->dyna.actor);
+    }
+}
+
+void BgKin2Bombwall_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgKin2Bombwall* this = (BgKin2Bombwall*)thisx;
+    this->actionFunc(this, globalCtx);
```

Comment:
```suggestion
    BgKin2Bombwall* this = (BgKin2Bombwall*)thisx;

    this->actionFunc(this, globalCtx);
```
We always have a blank line between declarations and code.